ui/packages/catalog-ui-search/src/main/webapp/component/source-item/source-item.view.js

```diff
@@ -13,23 +13,45 @@
  *
  **/
 /*global require*/
-var Marionette = require('marionette');
-var CustomElements = require('js/CustomElements');
-var template = require('./source-item.hbs');
+const Marionette = require('marionette');
+const CustomElements = require('js/CustomElements');
+const template = require('./source-item.hbs');
+const lightboxInstance = require('component/lightbox/lightbox.view.instance');
+const SourceAppView = require('component/source-app/source-app.view');
 
 module.exports = Marionette.ItemView.extend({
     template: template,
     tagName: CustomElements.register('source-item'),
     modelEvents: {
-        'change': 'render' 
+        'change': 'render'
     },
     events: {
-
+        'click button': 'onActionClicked'
+    },
+    regions: {
+        sourceApp: '.source-app'
     },
-    initialize: function(){
+    initialize(){
         
     },
-    onRender: function(){
+    onActionClicked(event) {
+
+        var target = event.currentTarget;
+
+        if(target !== null) {
+            const buttonId = target.getAttribute('data-id');
+            const clickedSourceAction = this.model.get('sourceActions').find((sourceAction) => sourceAction['title'] === buttonId);
```

Comment:
Are we sure that `title` will be unique among various source actions?

The two options I see here are:
- Split actions into their own views rather than unfurling them as part of this view.

OR

- Leave it as is but include all the data we need to open the lightbox as part of the element (as other data attributes).  

The second one is less work and fairly easy.  The first one is more if we plan on having more complicated source actions in the future.